service/history/workflow/update/registry.go

```diff
@@ -406,6 +419,20 @@ func (r *registry) checkInflightLimit() error {
 	return nil
 }
 
+func (r *registry) checkRegistrySizeLimit(payloadBytes int) error {
+	maxRegistrySize := r.maxRegistrySizeLimit()
+	if maxRegistrySize == 0 {
+		// limit is disabled
+		return nil
+	}
+	registrySize := r.GetSize()
```

Comment:
This will actually emit the existing metric `workflow_update_registry_size`; which is currently never emitted.